Sources/Scheduler.swift

```diff
@@ -162,11 +159,17 @@ public final class UIScheduler: Scheduler {
 		// already enqueued, we can skip scheduling and just execute directly.
 		if queued == 1 && DispatchQueue.getSpecific(key: UIScheduler.dispatchSpecificKey) == UIScheduler.dispatchSpecificValue {
 			actionAndDecrement()
+			return nil
 		} else {
-			DispatchQueue.main.async(execute: actionAndDecrement)
-		}
+			let disposable = AnyDisposable()
 
-		return disposable
+			DispatchQueue.main.async {
+				guard !disposable.isDisposed else { return }
```

Comment:
It was there before.
https://github.com/ReactiveCocoa/ReactiveSwift/pull/439/commits/04b4ddf4500a2f93dcbef286ace2d899c0918621#diff-f045af08e7f58fd005590d376ccd1215L141